vent/controller/control_module.py

```diff
@@ -147,21 +147,32 @@ def get_sensors(self) -> SensorValues:
 
     def get_alarms(self) -> List[Alarm]:
         # Returns all alarms as a list
+        lock.acquire()
         ls = self.logged_alarms
         for alarm_key in self.active_alarms.keys():
             ls.append(self.active_alarms[alarm_key])
+        lock.release()
         return ls
 
     def get_active_alarms(self):
         # Returns only the active alarms
-        return self.active_alarms
+        lock.acquire()
+        active_alarms = self.active_alarms
+        lock.release()
+        return active_alarms
 
     def get_logged_alarms(self) -> List[Alarm]:
         # Returns only the inactive alarms
-        return self.logged_alarms
+        lock.acquire()
+        logged_alarms = self.logged_alarms
+        lock.release()
+        return logged_alarms
 
     def set_control(self, control_setting: ControlSetting):
         # Updates the control settings, and re-calculate the derived control variables 
+
+        lock.acquire()
+
         if control_setting.name == ControlSettingName.PIP:
```

Comment:
Any place these variables are accessed in other areas of the control loop also need to have a lock.